### ISSUE TITLE
Deoplete source 

### ISSUE THREAD

**Issue by petobens** [gaalcaras/ncm-R#10]
Do you plan to implement (or do you happen to know if there is) a deoplete source for r completion? Thanks!

**Comment by kadekillary** [comment]
Definitely would be nice, especially considering nvim-completion-manager is [no longer maintained](https://github.com/roxma/nvim-completion-manager).

**Comment by mschilli87** [comment]
Just cross-referencing the relevant issues to avoid the same discussion everywhere:
* https://github.com/jalvesaq/Nvim-R/issues/326
* https://github.com/Shougo/deoplete.nvim/issues/745

The second was actually mentioned in the first: https://github.com/jalvesaq/Nvim-R/issues/326#issuecomment-386109839

The whole discussion started here: https://github.com/jalvesaq/Nvim-R/issues/326#issuecomment-385804953

And the maintainance situation was already mentioned here: https://github.com/jalvesaq/Nvim-R/issues/326#issuecomment-387163852

**Comment by gaalcaras** [comment]
FIY, I just merged the feature branch into `master`, you should use that instead of the now deleted feature branch.

**Comment by petobens** [comment]
Maybe @Shougo can suggest some way of porting the ncm2 source into deoplete, personally I have no idea how easy/difficult that might be...

**Comment by Shougo** [comment]
Hm...  Current deoplete source interface is pretty simple.  You can read the code in deoplete. 

> Deoplete is obviously larger than ncm-R, but I don't use it myself. I have no idea whether it's actually more stable than ncm2 or whether Shougo sometimes introduces breaking changes.

I keep the compatibility if possible.  It is not guaranteed though.

**Comment by khughitt** [comment]
@Shougo Any suggestions for examples of other custom sources of a similar nature that one might use as a template / to get some insights into interfacing with deoplete?

**Comment by Shougo** [comment]
You should read deoplete source code before asking me...

https://github.com/Shougo/deoplete.nvim/blob/master/rplugin/python3/deoplete/source/around.py

**Comment by khughitt** [comment]
Fair enough. I likely won't have time to take a stab at this at least until Fall, but perhaps if no one else has implemented this source by then, I'll take a stab at it.
